plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/BigQueryMetadataGenerator.scala

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.monster.sbt
+
+import java.nio.file.Path
+
+import io.circe.jawn.JawnParser
+import io.circe.syntax._
+import org.broadinstitute.monster.sbt.model.ColumnType.{Optional, _}
+import org.broadinstitute.monster.sbt.model.bigquery.BigQueryColumn
+import org.broadinstitute.monster.sbt.model.{MonsterTable, SimpleColumn}
+import sbt._
+import sbt.internal.util.ManagedLogger
+import sbt.nio.file.{FileAttributes, FileTreeView}
+
+/** Utilities for generating BigQuery metadata from Monster table definitions. */
+object BigQueryMetadataGenerator {
+  private val jsonParser: JawnParser = new JawnParser()
+
+  type BigQueryTable = Seq[BigQueryColumn]
+
+  /**
+    * Generate BigQuery metadata files based on the table definitions
+    * located in a local directory.
+    *
+    * @param inputDir directory containing table definitions in our JSON format
+    * @param inputExtension file extension used for our table definitions
+    * @param outputDir directory where BQ metadata files should be written
+    * @param fileView utility which can inspect the local filesystem
+    * @param logger utility which can write logs to the sbt console
+    */
+  def generateMetadata(
+    inputDir: File,
+    inputExtension: String,
```

Comment:
They're build settings right now, in case we wanted to store structs and tables in the same directory. I could simplify across the board by hard-coding required extensions.

Reply:
Gotcha! Sounds fine to me then.